sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -15,35 +18,94 @@ internal sealed class AzureSearchDocumentsEventSource : EventSource
         internal const int BatchSubmittedEvent = 2;
         internal const int RetryQueueResizedEvent = 3;
         internal const int PendingQueueResizedEvent = 4;
+        internal const int PublisherDisposedWithPendingActionsEvent = 5;
+        internal const int DocumentsPublishedEvent = 6;
+        internal const int ActionNotificationEventHandlerExceptionThrownEvent = 7;
 
         private AzureSearchDocumentsEventSource() :
             base(EventSourceName, EventSourceSettings.Default, AzureEventSourceListener.TraitName, AzureEventSourceListener.TraitValue)
         { }
 
         public static AzureSearchDocumentsEventSource Instance { get; } = new AzureSearchDocumentsEventSource();
 
-        [Event(PendingQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher pending queue size = {0}.")]
-        public void PendingQueueResized(int queueSize)
+        [Event(PendingQueueResizedEvent, Level = EventLevel.Verbose, Message = "{0}: pending queue size = {1}.")]
+        public void PendingQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(PendingQueueResizedEvent, queueSize);
+            WriteEvent(PendingQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher retry queue size = {0}.")]
-        public void RetryQueueResized(int queueSize)
+        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "{0}: retry queue size = {1}.")]
+        public void RetryQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(RetryQueueResizedEvent, queueSize);
+            WriteEvent(RetryQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has submitted batch of size {1}.")]
-        public void BatchSubmitted(string endPoint, int batchSize)
+        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "{0} at {1} has submitted batch of size {2}.")]
+        public void BatchSubmitted(string publisherType, string endPoint, int batchSize)
```

Comment:
Why do we include the endpoint on some of these but not others?

Reply:
Events fired from `Publisher` don't have the end point information.

Reply:
We should fix that next release so these read consistently.  We can add a `protected abstract Uri GetEndpoint()` in `Publisher` that `SearchIndexingPublisher` has to implement.

Reply:
Done.